mutable_tree.go

```diff
@@ -74,34 +71,32 @@ func (tree *MutableTree) IsEmpty() bool {
 
 // VersionExists returns whether or not a version exists.
 func (tree *MutableTree) VersionExists(version int64) bool {
-	tree.mtx.Lock()
-	defer tree.mtx.Unlock()
-
-	if tree.allRootLoaded {
-		return tree.versions[version]
+	firstVersion, err := tree.ndb.getFirstVersion()
```

Comment:
is there not a need to lock here anymore?

Reply:
yeah, we don't use the map of versions,since we are keeping the range of versions (fristVersion, latestVersion)